### PR TITLE
Don't push mined blocks to the chain during sync

### PR DESCRIPTION
**Pull request by ignopeverell**
Related to #1632

### CODE REVIEW

File: servers/src/mining/stratumserver.rs [hunk 1]
```diff
@@ -502,7 +502,7 @@ impl StratumServer {
 			return Err(serde_json::to_value(e).unwrap());
 		}
 		// If the difficulty is high enough, submit it (which also validates it)
-		if share_difficulty >= self.current_difficulty {
+		if share_difficulty >= self.current_difficulty && !self.sync_state.is_syncing() {
```
Comment (garyyu) [inline servers/src/mining/stratumserver.rs]: Could you please hold this PR for a while? 😄   to give us some time to address that 'impossible' happening found in #1632, and confirm that problem is well fixed.
I'm afraid if this fix is merged, then not easy to reproduce that root cause.

BTW, shall we log a warning message here? to remind the reason of this ignored 'submit' block.

Comment (ignopeverell) [reply servers/src/mining/stratumserver.rs]: Good point, we can definitely wait a few days for this. But I think it's already pretty obvious from the logs when you mine a block while syncing is enabled, no?

Comment (garyyu) [reply servers/src/mining/stratumserver.rs]: > But I think it's already pretty obvious from the logs when you mine a block while syncing is enabled

Yes, fully agree. 

I mean the fix in this PR is good and necessary, but the root cause for that strange **impossible** happening (https://github.com/mimblewimble/grin/issues/1632#issuecomment-426188635 - `update_header_head` return `None` but `header_head` has been updated!) could be a major risk and deserve a deep checking.

Comment (garyyu) [reply servers/src/mining/stratumserver.rs]: I find it's quite difficult to reproduce that special scenario: a bunch of Headers coming and in the same time mined a new block. So giving up to try reproducing problem, and please go on to merge this fix :)

**Comment by ignopeverell** [comment]
@antiochp not sure what it'd be, mining on top of an old block is unlikely to be productive and in itself could be seen as detrimental to the chain.

**Comment by antiochp** [comment]
Just to clarify. My question was around a peer lying about a fork.

* Miner is fully sync'd
* Miner happily mining on latest block
* peer advertises new fork with greater cumulative difficulty
* Miner kicks into sync mode and stops mining
* turns out peer was lying about the fork

**Comment by ignopeverell** [comment]
@antiochp very good point, I filed #1654 to think about that some more.

**Comment by antiochp** [comment]
The whole header MMR can of worms has got me thinking about sync a lot recently...

**Comment by ignopeverell** [comment]
Some more major code removal in the works?

**Pull request merged**